ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -527,15 +527,25 @@ removeSelection
   -> IO ()
 removeSelection acidDb WalletFace{..} walletSelRef runCardanoMode = do
   mWalletSel <- readIORef walletSelRef
+  walletDb <- query acidDb Snapshot
   newSelection <- case mWalletSel of
     Nothing -> pure Nothing
     -- Throw "Nothing selected" here?
     Just selection -> case selection of
       WSRoot hdrId -> do
+        let 
+          accList = (getAccounts (walletDb ^. dbHdWallets) hdrId) <&> (^. hdAccountId)
+          addrList = (getAddressesByRoot (walletDb ^. dbHdWallets) hdrId) <&> (^. hdAddressId)
+        -- delete the whole subtree. 
+        update acidDb (DeleteHdAddresses addrList)
+        update acidDb (DeleteHdAccounts accList)
         update acidDb (DeleteHdRoot hdrId)
```

Comment:
This is wrong, there should be only one `update`, because we want to delete the whole tree atomically. One `update` is done atomically, but three `update`s are not.
You should defined something like `DeleteHdTree` in `Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Delete` which will delete whole tree.
Something similar should be done for account deletion as well (there are two `update`s currently).